openverse_catalog/dags/providers/provider_dag_factory.py

```diff
@@ -302,7 +302,14 @@ def create_provider_api_workflow_dag(conf: ProviderWorkflow):
     )
 
     with dag:
-        ingest_data, ingestion_metrics = create_ingestion_workflow(conf)
+        # @stacimc is this stretching your beautiful default class a bit too far?
```

Comment:
:) I think it's a great idea as implemented!

Interestingly, I feel like the ingestion workflow isn't _that_ different from the normal one. Or at any rate, there's enough duplication that it's got me thinking of ways we could reuse the logic in `provider_dag_factory` -- especially since it's easy to imagine that we'll eventually add more providers that use this workflow.

This is more the case on `main` after #914, which broke out the `load_from_s3` task into smaller tasks including ones that could be reused here. The `create_loading_table`, `clean_data`, `upsert_data`, `drop_loading_table`, and `report_load_completion` tasks could all be reused, I think.

It would be cool to make `create_ingestion_workflow` a little more configurable by allowing individual steps to be overridden or skipped.  We have several other DAGs right now that are having to 'batch' their ingestion by splitting up the ingestion day into time intervals, that I think something like the dynamic tasks might even be helpful for our 'normal' workflows.